src/components/vapp.ts

```diff
@@ -0,0 +1,263 @@
+import * as paper from 'paper';
+import { EntityLabelComponent } from './entity-label';
+import { VmData } from './vm';
+import { VmAndVnicListComponent } from './vm-and-vnic-list';
+import { VAPP_BACKGROUND_COLOR } from '../constants/colors';
+import { CONNECTOR_RADIUS, VIEW_BOTTOM_PADDING, DEFAULT_SCROLLBAR_THICKNESS, VAPP_NETWORK_RIGHT_MARGIN, VAPP_PADDING,
+  DEFAULT_STROKE_WIDTH } from '../constants/dimensions';
+import { MarginComponent } from './margin';
+import { VappNetworkData } from './vapp-network';
+import { VappNetworkListComponent } from './vapp-network-list';
+import { ScrollbarComponent } from './scrollbar';
+
+const MARGIN_RIGHT = 30;
+const BACKGROUND_RADIUS = 5;
+const LABEL_ICON_COLOR = '#CA67B8';
+const VAPP_LABEL_BOTTOM_MARGIN = 10;
+const EDGE_LABEL_BOTTOM_MARGIN = 5;
+
+/**
+ * Interface for vApp data.
+ */
+export interface VappData {
+  uuid: string;
+  name: string;
+  vapp_networks: VappNetworkData[];
+  vms: VmData[];
+}
+
+/**
+ * Virtual Application Visual Component.
+ */
+export class VappComponent extends paper.Group {
+
```

Comment:
this is the major parent component that encapsulates everything else and handles most of the interdependent and shared data